openpype/hosts/unreal/plugins/publish/validate_sequence_frames.py

```diff
@@ -39,8 +39,20 @@ def process(self, instance):
             collections, remainder = clique.assemble(
                 repr["files"], minimum_items=1, patterns=patterns)
 
-            assert not remainder, "Must not have remainder"
-            assert len(collections) == 1, "Must detect single collection"
+            if remainder:
+                raise ValueError(
+                    "Some files have been found outside a sequence. "
+                    f"Invalid files: {remainder}")
+            if not collections:
+                raise ValueError(
+                    "No collections found. There should be a single "
+                    "collection per representation.")
+            if len(collections) > 1:
+                raise ValueError(
+                    "Multiple collections detected. There should be a single "
+                    "collection per representation. "
+                    f"Collections identified: {collections}")
+
```

Comment:
Use `raise PublishValidationError` so it formats nicely as a report.

Also, the "No collections found" I must admit would be quite hard to debug for an artist. What does it mean? What can I do about it? Where should be I looking? It's not mentioning what files it _did_ find either. We should provide artist facing reports that when seeing the logs with the description on the report the artist knows "I likely need to fix X or Y".

Reply:
Thanks for the feedback :) I've changed the exception and updated the message for not finding any collection! What do you think?

Reply:
Thanks - it is an improvement. I still feel like overall more care should be taken that the publish report is visually appealing for the artist, e.g. lists are now shown to artists as `[file1, file2, file3]` but might make more sense for artists as:
```
- file1
- file2
- file3
```
Nonetheless, the information is there so I suppose from here on out its "cosmetics" and not hard production requirement to fix now.

---

Regarding **designing good publish reports** I feel like we need some visual examples in the documentation @mkolar together with some good pointers on how one should approach it?

E.g. should **description** come with its own header? should description hold per instance information (as far as I know the intent was description was static per plugin, but the logs would show instance specific information to guide the artist).

Should we create an issue somewhere on where we track "to do for documentation"?

Reply:
I'll resolve this conversation so we can merge this PR, probably the forum is the best place to discuss the topic further and the changes to how the issues are presented to the artists can be implemented across all the DCCs in a dedicated PR.